### PR TITLE
Add Model.Meta.index_together fixer

### LINKED ISSUE #344: Auto create Meta.index values for index_together usages
### Description

Django 5.1 will be removing `index_together` in favor of `Meta.indexes`. It would be useful for this tool to either update the code or provide the suggested index to be used. Ideally the new `Index` would not change the database schema. It would only change the Django migration schema.

### PR DESCRIPTION
**Pull request by adamchainz**
Fixes #344.

Handles both forms of `index_together`, in list and tuple, and with/without existing `indexes` definition. Tried to cover every edge case I could think of, per tests.

Tested on Sentry, the only project I could find with historical `index_together` definitions. Use this commit: https://github.com/getsentry/sentry/commit/a32270c8ab84d341d97e15e95e0fa6832ae0a17c, just before github.com/getsentry/sentry/commit/00884d133eaaedf3e69bc8ab3d3bca781e0c425f and later commits updated `index_together`. 

Saw successful rewrites:

```console
$ git ls-files -z '*.py' | xargs -0 django-upgrade --target-version 4.2 --only index_together
Rewriting src/sentry/feedback/models.py
Rewriting src/sentry/incidents/models.py
Rewriting src/sentry/models/activity.py
Rewriting src/sentry/models/artifactbundle.py
Rewriting src/sentry/models/commit.py
Rewriting src/sentry/models/debugfile.py
Rewriting src/sentry/models/eventattachment.py
Rewriting src/sentry/models/group.py
Rewriting src/sentry/models/grouphistory.py
Rewriting src/sentry/models/groupinbox.py
Rewriting src/sentry/models/grouprelease.py
Rewriting src/sentry/models/organizationmembermapping.py
Rewriting src/sentry/models/outbox.py
Rewriting src/sentry/models/pullrequest.py
Rewriting src/sentry/models/release.py
Rewriting src/sentry/models/releasefile.py
Rewriting src/sentry/models/releaseprojectenvironment.py
Rewriting src/sentry/models/rule.py
Rewriting src/sentry/models/statistical_detectors.py
Rewriting src/sentry/models/userreport.py
Rewriting src/sentry/replays/models.py
```

...and the diff looks good:

```diff
$ git diff '*.py' | head -n 50
diff --git src/sentry/feedback/models.py src/sentry/feedback/models.py
index 3527d584bc4..b15d3a8ba45 100644
--- src/sentry/feedback/models.py
+++ src/sentry/feedback/models.py
@@ -27,6 +27,6 @@ class Feedback(Model):
     class Meta:
         app_label = "feedback"
         db_table = "feedback_feedback"
-        index_together = [("project_id", "date_added")]
+        indexes = [models.Index(fields=("project_id", "date_added"))]

     __repr__ = sane_repr("project_id", "feedback_id")
diff --git src/sentry/incidents/models.py src/sentry/incidents/models.py
index 6a4f4f6ae39..06e84dc7136 100644
--- src/sentry/incidents/models.py
+++ src/sentry/incidents/models.py
@@ -205,7 +205,7 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_incident"
         unique_together = (("organization", "identifier"),)
-        index_together = (("alert_rule", "type", "status"),)
+        indexes = [models.Index(fields=("alert_rule", "type", "status"))]

     @property
     def current_end_date(self):
@@ -561,7 +561,7 @@ class Meta:
         app_label = "sentry"
         db_table = "sentry_incidenttrigger"
         unique_together = (("incident", "alert_rule_trigger"),)
-        index_together = (("alert_rule_trigger", "incident_id"),)
+        indexes = [models.Index(fields=("alert_rule_trigger", "incident_id"))]


 class AlertRuleTriggerManager(BaseManager["AlertRuleTrigger"]):
diff --git src/sentry/models/activity.py src/sentry/models/activity.py
index aa9e0545adc..57575e9ac6b 100644
--- src/sentry/models/activity.py
+++ src/sentry/models/activity.py
@@ -109,7 +109,7 @@ class Activity(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_activity"
-        index_together = (("project", "datetime"),)
+        indexes = [models.Index(fields=("project", "datetime"))]

     __repr__ = sane_repr("project_id", "group_id", "event_id", "user_id", "type", "ident")

diff --git src/sentry/models/artifactbundle.py src/sentry/models/artifactbundle.py
index 323912d6d98..cbfb61c8805 100644
--- src/sentry/models/artifactbundle.py
```

One model was missed due to no `from django.db import models` in the file - I am gonna leave that for extension, if anyone wants.

### CODE REVIEW

**Comment by adamchainz** [comment]
@tim-schilling @UnknownPlatypus I would appreciate any pre- or post-merge code review, but no pressure.

**Comment by adamchainz** [comment]
@David-Wobrock As the author of the change to Django (two years ago!), perhaps you’d like to have a peek? 😇

**Review by UnknownPlatypus — APPROVED** [review]

**Comment by UnknownPlatypus** [comment]
Not much to say, looking good and the results on sentry are looking good also! 
I tried it on a few personal repo and it worked as expected on the few matches!

File: src/django_upgrade/fixers/index_together.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+) -> Iterable[tuple[Offset, TokenFunc]]:
+    if (
+        node.name != "Meta"
+        or sum(isinstance(p, ast.ClassDef) for p in parents[1:]) != 1
+        or not all(isinstance(subnode, ast.Assign) for subnode in node.body)
+    ):
+        return
+
+    # Find rewritable index_together declaration
+    index_togethers: list[ast.Assign] = []
```
[the remark this answers is not in the document]
Comment (adamchainz) [reply src/django_upgrade/fixers/index_together.py:53]: Yes, exactly, just in case.

**Comment by adamchainz** [comment]
Thank you for testing!

File: tests/fixers/test_index_together.py [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+                ]
+        """,
+        """\
+        from django.db import models
+
+        class Duck(models.Model):
+            class Meta:
+                indexes = [
+                    models.Index("bill"),
+                models.Index(fields=("bill", "tail"))]
```
Comment (David-Wobrock) [inline tests/fixers/test_index_together.py:341]: Funky indentation here and below 😛 

But any reasonable project that uses a formatter, may it by black or ruff, should be fine :)

**Review by David-Wobrock — APPROVED** [review]
Looks pretty good 👍

File: tests/fixers/test_index_together.py (hunk 2 above)
Comment (adamchainz) [reply tests/fixers/test_index_together.py:341]: Yeah, the README explicitly says:

> django-upgrade focuses on upgrading your code and not on making it look nice. Run django-upgrade before formatters like Black.

It’s hard enough to ensure generated code is valid, nice formatting is an extra level and just duplicative of black etc. anyway.
